src/main/java/org/fenixedu/academic/domain/degreeStructure/ProgramConclusionConfig.java

```diff
@@ -0,0 +1,149 @@
+package org.fenixedu.academic.domain.degreeStructure;
+
+import java.util.Collection;
+
+import org.fenixedu.academic.domain.DegreeCurricularPlan;
+import org.fenixedu.academic.domain.exceptions.DomainException;
+import org.fenixedu.bennu.core.domain.Bennu;
+import org.fenixedu.commons.i18n.LocalizedString;
+
+public class ProgramConclusionConfig extends ProgramConclusionConfig_Base {
+    
+    protected ProgramConclusionConfig() {
+        super();
+        super.setRootDomainObject(Bennu.getInstance());
+    }
+
+    protected void init(LocalizedString conclusionTitle, DegreeCurricularPlan degreeCurricularPlan,
+            ProgramConclusion programConclusion) {
+        setConclusionTitle(conclusionTitle);
+        setDegreeCurricularPlan(degreeCurricularPlan);
+        setProgramConclusion(programConclusion);
+        changeConfigOrder(degreeCurricularPlan.getProgramConclusionConfigsSet().size() - 1);
+    }
+
+    public static ProgramConclusionConfig create(LocalizedString conclusionTitle, DegreeCurricularPlan degreeCurricularPlan,
+            ProgramConclusion programConclusion) {
+        ProgramConclusionConfig config = new ProgramConclusionConfig();
+        config.init(conclusionTitle, degreeCurricularPlan, programConclusion);
+
+        return config;
+    }
+
+    public void edit(LocalizedString conclusionTitle, ProgramConclusion programConclusion) {
+        setConclusionTitle(conclusionTitle);
+        setProgramConclusion(programConclusion);
+    }
+
+    @Override
+    public void setConclusionTitle(final LocalizedString conclusionTitle) {
+        if (conclusionTitle == null || conclusionTitle.isEmpty()) {
+            throw new DomainException("error.ProgramConclusionConfig.conclusionTitle.cannot.be.null.or.empty");
+        }
+        super.setConclusionTitle(conclusionTitle);
+    }
+
+    @Override
+    public void setDegreeCurricularPlan(final DegreeCurricularPlan degreeCurricularPlan) {
+        if (degreeCurricularPlan == null) {
```

Comment:
In DML the multiplicity of the relation with degree curricular plan is 1, so this check shouldn't be necessary.